code/script.js

```diff
@@ -229,97 +264,145 @@ const setSecret = () => {
+      alert(
+        `No, the person doesn't have ${value}! Remove all people with ${value}!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
+const guess = (suspectCharacter) => {
   // store the interaction from the player in a variable.
-  // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmGuess = confirm(`Are you sure you want to guess?`)
+  if (confirmGuess)
+    checkMyGuess(suspectCharacter)
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+const checkMyGuess = (suspectCharacter) => {
+  if (suspectCharacter === secret.name) {
+    winOrLoseText.innerHTML = `Hurray, you win! Congratulations!`
+    gameWinSound.play()
+  } else {
+    winOrLoseText.innerHTML = `Baw, you lose...it was ${secret.name}!`
+    gameLoseSound.play()
+  }
+  winOrLose.style.display = 'flex'
+  boardWrapper.style.display = 'none'
 }
 
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+startGameButton.addEventListener('click', () => {
+  start()
+  startGame.style.display = 'none'
+  boardWrapper.style.display = 'flex'
+  gameStartSound.play()
+})
+restartButton.addEventListener('click', () => {
+  start()
+  startGame.style.display = 'none'
+  boardWrapper.style.display = 'flex'
+})
+questions.addEventListener('change', (event) => {
+  selectQuestion(event.target.value)
+})
+filterButton.addEventListener('click', () => {
+  checkQuestion()
+  count += 1
+  countGuess.innerHTML = "Guesses:" + count
+})
+
+playAgainButton.addEventListener('click', () => {
+  start()
+  startGame.style.display = 'none'
+  winOrLose.style.display = 'none'
+  boardWrapper.style.display = 'flex'
+  gameStartSound.play()
```

Comment:
maybe put gameLoseSound.pause() and gameWinSound.pause() so the sounds don't clash when user clicks play again button quickly